src/ui_basic/dropdown.h

```diff
@@ -275,9 +287,78 @@ template <typename Entry> class Dropdown : public BaseDropdown {
 	                  type,
 	                  style,
 	                  button_style) {
+		set_handle_textinput();
 	}
 	~Dropdown() override {
-		entry_cache_.clear();
+		filtered_entries.clear();
+		unfiltered_entries.clear();
+	}
+
+	void clear_filter() override {
+		if (current_filter_.empty()) {
+			return;
+		}
+		current_filter_.clear();
+		if (has_selection()) {
+			Entry selected_entry = get_selected();
+			restore_filtered_list();
+			select(selected_entry);
+		} else {
+			restore_filtered_list();
+		}
+	}
+	void restore_filtered_list() {
+		clear_filtered_list();
+		for (auto& ee : unfiltered_entries) {
+			add_to_filtered_list(ee.name, ee.value, ee.img, false, ee.tooltip, ee.hotkey);
+		}
+	}
+
+	void save_selected_entry(uint32_t index) override {
+		selected_entry_ = index < filtered_entries.size() ? *filtered_entries[index] : Entry{};
+	}
+
+	bool handle_textinput(const std::string& input_text) override {
+		std::string lower = std::string(input_text);
+		lower[0] = std::tolower(lower[0]);
```

Comment:
This could potentially contain multiple characters. Better iterate over the whole string and replace every index with its lowercase equivalent.

Reply:
Also once you're using `tolower` and once `std::tolower`…?